src/reconcile/rec_visibility.c

```diff
@@ -272,13 +272,18 @@ static int
 __rec_find_and_save_delete_hs_upd(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_INSERT *ins,
```

Comment:
May be rename this function as we are doing something more here?

Reply:
Yes, changed to `__rec_save_delete_hs_upd_and_free_obs_updates` to include freeing the obsolete updates.